### DISCUSSION TITLE
Add a way to document model attributes with description, examples, etc.

### DISCUSSION THREAD

**Discussion by sNok3** [dedoc/scramble#1024] [Ideas]
As the title suggests, when generating the docs for a controller which has model binding, the schema for the model will result in just the fields that the model has, without the annotations it has.

**Comment by romalytvynenko** [comment]
@sNok3 please share the example of your code and resulting docs. Also would be great to understand what your expected

**Comment by sNok3** [comment]
<img width="610" alt="image" src="https://github.com/user-attachments/assets/6392aace-c8d6-4ea6-82c7-bce24a47316e">

```php
class Driver extends Model
{
    use HasFactory;
    use HasStatuses;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        /**
         * The internal ID of the associated route
         *
         * @var int
         *
         * @example 1
         */
        'route_id',
        /**
         * The identifier of the activity
         *
         * @example 3d4b6f0e-2b0b-4b0b-8b4b-0b2b4b0b2b4b
         */
        'identifier'
   ];
```
So as you can see, I have annotations for both fields here, but under the schemas, there are is no information about the model.
<img width="320" alt="image" src="https://github.com/user-attachments/assets/53faef1b-cf74-4b17-836a-36206826bf1b">

**Comment by romalytvynenko** [comment]
@sNok3 ooh, gotcha. So basically you want to annotate the model fields with descriptions and examples? 

Currently this is not supported. 

You can use API resources and there you can add the descriptions.

However, I'll keep this issue open to add this possibility in future.

**Comment by sNok3** [comment]
Yeah, that is exactly what I needed. However, using API resources in the controller method isn't really my way to go for now since I have a big codebase.

However, I'll gladly wait for this to get implemented!

**Comment by alecpl** [comment]
At least type and description could be fetched from the `@property` definitions on the model class, please.

**Comment by romalytvynenko** [comment]
@chrisvanlier2005 as far as I remember, you've tried implementing `@property` support on model classes in your Scramble fork?

**Reply by chrisvanlier2005** [reply]
@romalytvynenko That's correct. I've implemented a PropertyTypesFromPhpdocExtension (private repo) that retrieves PHPDoc types from `@property`, `@property-read`, and `@var` tags on properties, as well as promoted `@param` tags.
However, this extension only handles type fetching... descriptions and other annotations aren't currently supported. In my opinion, those should be reserved for the API Resource layer instead

If there is interest, i'd be willing to submit a PR 🙂

**Reply by romalytvynenko** [reply]
@chrisvanlier2005 please go ahead, I'll be glad to add descriptions support on top of your extension if needed

**Reply by chrisvanlier2005** [reply]
Awesome, i'll submit PR when I have time 🙂
